Adapter/Configuration.php

```diff
@@ -25,8 +25,22 @@
  */
 namespace PrestaShop\PrestaShop\Adapter;
 
-class Configuration implements \PrestaShop\PrestaShop\Core\ConfigurationInterface
+use PrestaShop\PrestaShop\Core\Foundation\Exception;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
 {
+    private $shop;
+
+    public function __construct(\Shop $shop = null)
+    {
+        if (is_null($shop)) {
```

Comment:
I think it will be the same since I typehinted Shop: `__construct(\Shop $shop = null)`
If you pass something else, it will fail.